### PR TITLE
Milestone four

### PR DESCRIPTION
**Pull request by expjazz**
As requested, we implemented the filter so the user can see the book according to the category filter/select. To do so, we implemented a filter in the store. The component that implements the filter is the Category Filter. The conditional method is a selector in the selectors folder.

### CODE REVIEW

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @expjazz @mariosknl  
You have done a great job on this milestone 👏👏 however, there are a few things that we need to fix.
**See my comment below**
## Status: Changes Required ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions or need any help_

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export default (state = {}, action) => {
+  switch (action.type) {
+    case ('CHANGE_FILTER'):
+      return ({
+        ...state,
+        current: action.payload,
+      });
+    default:
+      return state;
+  }
```
Comment (fatymahmed) [inline src/reducers/filter.js:12]: - It should default to `All`  see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/HandelChangeFilter.png) from Pathwright

File: src/components/CategoryFilter.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const categories = useSelector(state => state.filter.categories);
+
+  return (
+    <CategoryFilterSelect onChange={handleChange}>
+      <option value="All">All</option>
+      {categories.map(category => {
+        count += 1;
+        return <option value={category} key={count}>{category}</option>;
+      })}
+      <option>Horror</option>
```
Comment (fatymahmed) [inline src/components/CategoryFilter.js:24]: [OPTIONAL]

why do you have this here?  
The only category that shouldn't be in the categories list is `all` see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/AllReturnsAllCategories.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896829/)

**Review by shshamim63 — APPROVED** [review]
## :+1: Approved
___
Hi @expjazz and @mariosknl :wave: ,
Awesome job, you have successfully completed all the changes asked by the previous reviewer. Just got a simple suggestion to make. You can update your readme a bit more with the info related to set up the project in a dev end. 
___
#### This :+1: is from me, you can go ahead and merge it.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Pull request closed without merging**